accounts/abi/bind/backends/simulated.go

```diff
@@ -350,6 +350,18 @@ func (b *SimulatedBackend) CallContract(ctx context.Context, call ethereum.CallM
 		return nil, err
 	}
 	res, err := b.callContract(ctx, call, b.blockchain.CurrentBlock(), state)
+
+	if res != nil && res.Err != vm.ErrOutOfGas {
```

Comment:
out of gas is handled in the original flow, right?

Reply:
if `err == ErrOutOfGas` then it's not a revert :)